### ISSUE TITLE
Can't create product image attribute programmatically

### ISSUE THREAD

**Issue by Axel29** [magento/magento2#23245]
### Preconditions

1. Magento 2.3.0
2. PHP 7.2
3. MySQL 5.7

### Steps to reproduce

1. Install Magento 2.3.0
2. Create custom product attribute programmatically and add it to "Images" group
```php
<?php
namespace Custom\Catalog\Setup;

use Magento\Catalog\Model\Product\Attribute\Frontend\Image as ImageFrontendModel;
use Magento\Eav\Model\Entity\Attribute\ScopedAttributeInterface;
use Magento\Eav\Model\Entity\Attribute\Source\Boolean;
use Magento\Eav\Setup\EavSetupFactory;
use Magento\Framework\Setup\InstallDataInterface;
use Magento\Framework\Setup\ModuleContextInterface;
use Magento\Framework\Setup\ModuleDataSetupInterface;

class InstallData implements InstallDataInterface
{
    /** @var \Magento\Eav\Setup\EavSetupFactory $_eavSetupFactory */
    protected $_eavSetupFactory;

    /**
     * InstallData constructor.
     *
     * @param \Magento\Eav\Setup\EavSetupFactory $eavSetupFactory
     */
    public function __construct(EavSetupFactory $eavSetupFactory)
    {
        $this->_eavSetupFactory = $eavSetupFactory;
    }

    /**
     * {@inheritDoc}
     */
    public function install(ModuleDataSetupInterface $setup, ModuleContextInterface $context)
    {
        $setup->startSetup();

        $eavSetup = $this->_eavSetupFactory->create(['setup' => $setup]);

        $eavSetup->addAttribute(
            \Magento\Catalog\Model\Product::ENTITY,
            'pictogram',
            [
                'type'                    => 'varchar',
                'label'                   => 'Pictogram',
                'input'                   => 'media_image',
                'frontend'                => ImageFrontendModel::class,
                'required'                => false,
                'sort_order'              => 2,
                'global'                  => ScopedAttributeInterface::SCOPE_STORE,
                'used_in_product_listing' => true,
                'group'                   => 'Images',
                'source'                  => Boolean::class,
                'visible'                 => true,
                'user_defined'            => true,
                'visible_on_front'        => true,
            ]
        );

        $setup->endSetup();
    }
}

```
3. Run `php bin/magento setup:upgrade` command in terminal

### Expected result

1. The `setup:upgrade` command should finish correctly
2. The attribute should be created

### Actual result

1. A SQL integrity constraint is triggered on the `setup:upgrade` command:
```
SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '4-Images' for key 'EAV_ATTRIBUTE_GROUP_ATTRIBUTE_SET_ID_ATTRIBUTE_GROUP_NAME', query was: INSERT INTO `eav_attribute_group` (`attribute_set_id`, `attribute_group_name`, `sort_order`, `attribute_group_code`) VALUES (?, ?, ?, ?)
```
![image](https://user-images.githubusercontent.com/34167293/59449602-b5f29500-8e07-11e9-87de-781751b62919.png)
2. The attribute is not created

### Additional information

The bug seems to be caused by a mismatch between attribute group "Images" code and name.
In the table "eav_attribute_group", the column "attribute_group_name" is equal to "Images" for product entity, but it's code (column "attribute_group_code") is registered to "image-management".

Thus, in the `\Magento\Eav\Setup\EavSetup::addAttribute` method, Magento calls  `\Magento\Eav\Setup\EavSetup::addAttributeGroup`, which is based on the "attribute_group_code" in database, which doesn't exist as "Images" and then tries to create it again.

### Workaround 

A temporary solution is to use "image-management" as group, but that causes a rename in "eav_attribute_group" of the attribute group...

### Attachments

Here's a [sample module](https://github.com/magento/magento2/files/3287140/Custom_Catalog_Sample.zip) to test the bug.

**Comment by Axel29** [comment]
Hi ,

It would be hard to use this vanilla Magento instance since it requires adding custom code.

**Comment by engcom-Charlie** [comment]
Hello @Axel29 ! Thank you for your report .The problem is that PHP 7.7 is not supported by Magento 2.3 but i still can reproduce your issue using PHP 7.2 .

**Comment by wclrz** [comment]
Sorry typo error, I meant PHP 7.2 :-)

**Comment by Axel29** [comment]
Sorry typo error, I meant PHP 7.2 :-)

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-qa-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-17495`, `MC-17496` were created

**Issue Available**: @engcom-qa-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magesanjay** [comment]
Hello @Axel29,
I have review your script.
There is no relation between media image and Group.
You have remove `'group' => 'Images'` from your InstallData script no any error at code side.

**Comment by Axel29** [comment]
@magesanjay I'm sorry but I don't understand what you mean?
The purpose of this issue is that the `'group' => 'Images'` breaks the script due to the database value in `attribute_group_code`.

The "media_image" input is just for the input type.

**Comment by magesanjay** [comment]
Hey @Axel29 
Yes exactly!

**Comment by choussamaster** [comment]
A quick fix 
`$eavSetup = $this->eavSetupFactory->create();

            $eavSetup->addAttribute(
                \Magento\Catalog\Model\Product::ENTITY,
                'color_image',
                [
                    'type' => 'varchar',
                    'label' => 'color_image',
                    'input' => 'media_image',
                    'required' => false,
                    'sort_order' => 30,
                    'frontend' => \Magento\Catalog\Model\Product\Attribute\Frontend\Image::class,
                    'global' => \Magento\Eav\Model\Entity\Attribute\ScopedAttributeInterface::SCOPE_STORE,
                    'used_in_product_listing' => true,
                    'user_defined' => true,
                    'visible' => true,
                    'visible_on_front' => true
                ]
            );
            $id = $eavSetup->getAttributeId(
                \Magento\Catalog\Model\Product::ENTITY,
                'color_image'
            );

            $attributeSetId = $eavSetup->getDefaultAttributeSetId(\Magento\Catalog\Model\Product::ENTITY);
            $eavSetup->addAttributeToGroup(\Magento\Catalog\Model\Product::ENTITY, $attributeSetId, 'image-management', $id, 10);`

**Comment by magento-engcom-team** [comment]
Hi @Axel29. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-23750` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.

**Comment by zahirabbasb** [comment]
its still produced in M2.4.4
